src/concurrent/files.rs

```diff
@@ -236,15 +238,15 @@ pub(crate) struct CovdirFilesWcc {
 
 impl CovdirFilesWcc {
```

Comment:
Can we deduplicate some of the code present in this function? We can use smaller functions and compose them